example-app/app/Models/Flight.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use App\Exceptions\{
+    ArrivalCityNotAssociatedException,
+    DepartureCityNotAssociatedException,
+    InvalidArrivalTimeException,
+    InvalidDepartureTimeException,
+    SameCityException
+};
+
+class Flight extends Model
+{
+    use HasFactory;
+
```

Comment:
You can cast the datetimes on db automatically to Carbon, so it's easier and less error prone to handle datetimes


https://laravel.com/docs/11.x/eloquent-mutators#date-casting
https://carbon.nesbot.com/docs/